flake.nix

```diff
@@ -0,0 +1,91 @@
+{
+  description = "ADO.net and JDBC Connection String Parser.";
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    rust-overlay = {
+      url = "github:oxalica/rust-overlay";
+      inputs.nixpkgs.follows = "nixpkgs";
+      inputs.flake-utils.follows = "flake-utils";
+    };
+  };
+
+  outputs = { self, nixpkgs, flake-utils, rust-overlay }:
+    flake-utils.lib.eachDefaultSystem (system: let
+      overlays = [ (import rust-overlay) ];
+      pkgs = import nixpkgs { inherit system overlays; };
+      rust = pkgs.rust-bin.fromRustupToolchainFile ./rust-toolchain.toml;
+      inherit (pkgs) wasm-bindgen-cli rustPlatform nodejs;
+    in {
+      defaultPackage = rustPlatform.buildRustPackage {
+        name = "connection-string";
+        src = builtins.path { path = ./.; name = "connection-string"; };
+
+        cargoLock = {
+          lockFile = ./Cargo.lock;
+        };
+
+        nativeBuildInputs = [ rust wasm-bindgen-cli nodejs ];
```

Comment:
nitpick: do we need nodejs as a build input here? I don't see where

Reply:
nope.

Reply:
removed it and added a devshell